ml-agents/mlagents/trainers/torch/networks.py

```diff
@@ -107,37 +107,40 @@ def forward(
         memories: Optional[torch.Tensor] = None,
         sequence_length: int = 1,
     ) -> Tuple[torch.Tensor, torch.Tensor]:
-        encodes = []
-        goal_signal = None
+        obs_encodes = []
+        goal_encodes = []
         for idx, processor in enumerate(self.processors):
             obs_input = inputs[idx]
             processed_obs = processor(obs_input)
             if (
                 self.obs_types[idx] == ObservationType.DEFAULT
                 or self.conditioning_type == ConditioningType.DEFAULT
             ):
```

Comment:
I would only check for the obs_type here
```suggestion
            if self.obs_types[idx] == ObservationType.DEFAULT:
```
And have a separate block of code that checks if `self.conditioning_type == ConditioningType.DEFAULT` and if it is, concatenate  obs and goals. It will be cleaner if obs_encode and goal_encodes are always the same regardless of conditioning type

Reply:
Like this?

```
            if self.obs_types[idx] == ObservationType.DEFAULT:
                obs_encodes.append(processed_obs)
            elif self.obs_types[idx] == ObservationType.GOAL:
                goal_encodes.append(processed_obs)
            else:
                raise Exception(
                    "TODO : Something other than a goal or observation was passed to the agent."
                )

        if self.conditioning_type == ConditioningType.DEFAULT:
            obs_encodes = obs_encodes + goal_encodes
```